api/model/src/test/java/org/projectnessie/model/types/CustomTestContentTypeBundle.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model.types;
+
+import org.projectnessie.model.types.ContentTypes.Registrar;
+
+public class CustomTestContentTypeBundle implements ContentTypeBundle {
```

Comment:
Suggestion: Do we really need the `ContentTypeBundle` interface? Why not load each custom type class via the service loader and then register them all automatically based on `@JsonTypeName` (which we validate to match the registered type name anyway)?

IIRC, service loader resources can list more than one service class per file.

With current approach it looks like type names leak across class boundaries without any good code-level reason :thinking: 

If that works, we might as well follow the same approach for `ContentSerializerBundle`, I guess.

Reply:
In theory, yes.
But we need `Class<? extends Content>` for the registration. Service loaders however return instantiated instances of the class.